`add-collaborators` tweaks

jaw6 opened the pull request:
A couple of tweaks to `add-collaborators` after using it in a particularly busy class last week:
- Grabs a hash of current collaborators before working through issue comments: skips anyone who is already a collaborator
  - Old version would still `post` to the collaborators API for _every_ comment in the issue
  - If there are 100s of comments, this should be considerably faster
- Posts a comment as teacher at the end of the script with `foo, bar, baz are now repository collaborators`
  - should cut down on student confusion - they should now get a notification & see a comment when they've been added
- One or two other minor things while I was in there. (mostly `abort` instead of just `puts` on error...)

@loranallensmith @brntbeer @patrickmckenna any feedback here? I've done some testing and confirmed it "works on my machine" -- got a minute to try this out?

jaw6 committed:
Store current collaborators

For large classes, this should speed up the script considerably.

jaw6 committed:
Add comment to issue, list new collaborators

Teacher posts comments (at end of script): 

"foo, bar, baz are now repository collaborators"

Or: 

"qux is now repository collaborator"

jaw6 committed:
exit on error

Skips comment if Octokit errors during add-collaborators

patrickmckenna commented on add-collaborators:
```diff
@@ -38,19 +38,35 @@ puts({repo_name: repo_name, issue_num: issue_num}.inspect)
+if successfully_added_users.any?
+  begin
+    verb = successfully_added_users.size == 1 ? "is" : "are"
+    message = ":tada: #{successfully_added_users.map { |name| "@#{name}" }.join(" ")} #{verb} now repository collaborator#{successfully_added_users.size == 1 ? '' : 's'} :balloon:"
```
@jaw6 Should have an indefinite article: "now _a_ repository collaborator" (instead of "now repository collaborator") when `successfully_added_users.size == 1`.

loranallensmith replied:
:+1: Just noticed this too!

jaw6 replied:
> Should have an indefinite article

Good catch. English is hard.

![screen shot 2015-10-06 at 4 20 16 pm](https://cloud.githubusercontent.com/assets/2077/10323064/6063ca1e-6c46-11e5-9500-3f8fd919e155.png)

patrickmckenna commented:
@jaw6 :+1: to all of this.

patrickmckenna commented:
Also, works on my machine, too.

loranallensmith commented:
@jaw6 This is so :cool:.  Thanks for making these updates.  I just ran this and it appears to be working on my machine.

![screen shot 2015-10-06 at 2 50 51 pm](https://cloud.githubusercontent.com/assets/3477155/10318942/b1c3042c-6c39-11e5-9ad3-5f171f81f7cd.png)

jaw6 committed:
Use more properful englishization hereabouts

"foo, bar, and baz are now repository collaborators"

"foo and bar are now repository collaborators"

"foo is now a repository collaborator"

loranallensmith commented:
:tada::balloon:

Just wanted to give another :+1: to this type of refinement.  I'm not sure if any other @github/training-teachers have experienced the same thing I have, but students regularly ask me during class if I would mind sharing how we do "tricks" like this.  Utilitarian as its roots may be, it's a really slick tool to have.  Students love knowing that they have such an accessible platform for automation at their disposal.

:muscle: @jaw6 and @patrickmckenna

brntbeer commented:
@jaw6 this is amazing. Looks great. Particularly like the use of Aborts instead of puts! :wink:

crichID commented:
@jaw6 I'm late to the party but this is so awesome! Thank you for making these updates.
